test/gtest/gpu_nchw_nhwc_transpose.cpp

```diff
@@ -0,0 +1,658 @@
+
+                                k(opArgs_2d);
+                            };
+                        });
+
+                    std::vector<miopen::solver::KernelInfo> construction_params_3d{
+                        transpose_sol_3d.GetKernelInfo()};
+                    std::vector<miopen::solver::KernelInfo> construction_params_2d{
+                        transpose_sol_2d.GetKernelInfo()};
+
+                    const auto invoker_3d =
+                        miopen::deref(this->handle).PrepareInvoker(*invoker_factory_3d, construction_params_3d);
+                    const auto invoker_2d =
+                        miopen::deref(this->handle).PrepareInvoker(*invoker_factory_2d, construction_params_2d);
+
+                    const auto invoke_param_3d = transpose_invoke_param{
+                        DataCast(static_cast<const void*>(src_3d_dev)), DataCast(dst_3d_dev)};
+                    const auto invoke_param_2d = transpose_invoke_param{
+                        DataCast(static_cast<const void*>(src_2d_dev)), DataCast(dst_2d_dev)};
+
+                    // run gpu
+                    invoker_3d(miopen::deref(this->handle), invoke_param_3d);
+                    invoker_2d(miopen::deref(this->handle), invoke_param_2d);
+
+                    ASSERT_HIP_SUCCESS(hipMemcpy(
+                        t_gpu_3d.data.data(), dst_3d_dev, sizeof(T) * tensor_sz, hipMemcpyDeviceToHost));
+                    ASSERT_HIP_SUCCESS(hipMemcpy(
+                        t_gpu_2d.data.data(), dst_2d_dev, sizeof(T) * tensor_sz, hipMemcpyDeviceToHost));
+
+                    // run cpu
+                    cpu_transpose<T, TRANSPOSE_SOL>::run(t_dst_ref.data.data(), t_src.data.data(), n, c, d, h, w);
+
+#ifdef BREAK_IT
+        // TEMPCODE break it
+        t_dst_ref[prng::gen_0_to_B(tensor_sz)] += 1;
+#endif
+
+                    // we expect exact match, since use integer
+                    verify_tensor(t_gpu_3d, "gpu3d", t_dst_ref, "cpu");
+                    verify_tensor(t_gpu_2d, "gpu2d", t_dst_ref, "cpu");
+                }
+            }
+        }
+    }
+    virtual void TearDown() override
+    {
+        hipFree(src_3d_dev);
+        hipFree(dst_3d_dev);
+        hipFree(src_2d_dev);
+        hipFree(dst_2d_dev);
+
+        miopenDestroy(handle);
+    }
+};
+
+#define DEFINE_TransposeTest_2D(type, sol)                                                          \
+struct TransposeTest_2D_ ## sol ## _ ## type                                                        \
+    : public TransposeTest_2D<type, miopen::sol>                                                    \
+{                                                                                                   \
+protected:                                                                                          \
+    void SetUp() override { TransposeTest_2D<type, miopen::sol>::SetUp(); }                         \
```

Comment:
why do you need to `override` these methods if you're only calling the base class versions.

Reply:
Good catch. I just didn't trust gtest and wanted to be certain they get called. The overrides were not needed.